specs/proposals.md

```diff
@@ -47,43 +49,31 @@ output, and the appropriate timestamp, to the [L2 Output Commitment](#l2-output-
 
 ## L2 Output Commitment Construction
 
-This merkle-structure is defined with [SSZ], a type system for merkleization and serialization, used in
-L1 (beacon-chain). However, we replace `sha256` with `keccak256` to save gas costs in the EVM.
-
-[SSZ]: https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md
+The `output_root` is a 32 byte string defined as:
 
-```python
-class L2Output(Container):
-  state_root: Bytes32
-  withdrawal_storage_root: Bytes32  # TODO: withdrawals specification work-in-progress
-  latest_block: ExecutionPayload  # includes block hash
-  history_accumulator_root: Bytes32  # Not functional yet
-  extension: Bytes32
+```pseudocode
+keccak256(version_byte . state_root . withdrawal_storage_root . latest_block)
 ```
 
-The `state_root` is the Merkle-Patricia-Trie ([MPT][g-mpt]) root of all execution-layer accounts,
-also found in `latest_block.state_root`: this field is frequently used and thus elevated closer to the L2 output root,
-as opposed to retrieving it from the pre-image of the block in `latest_block`,
-reducing the merkle proof depth and thus the cost of usage.
+where the concatenated variables are:
 
-The `withdrawal_storage_root` elevates the Merkle-Patricia-Trie ([MPT][g-mpt]) root of L2 Withdrawal contract storage.
-Instead of a MPT proof to the Withdrawal contract account in the account trie,
-one can directly access the MPT storage trie root, thus reducing the verification cost of withdrawals on L1.
+1. The `version_byte` (`bytes32`) a simple version string which increments anytime the construction of the output root
+   is changed.
 
-The `latest_block` is an execution-layer block of L2, represented as the [`ExecutionPayload`][ExecutionPayload] SSZ type
-defined in L1. There may be multiple blocks per L2 output root, only the latest is presented.
+1. The `state_root` (`bytes32`) is the Merkle-Patricia-Trie ([MPT][g-mpt]) root of all execution-layer accounts,
+   also found in `latest_block.state_root`: this field is frequently used and thus elevated closer to the L2 output
+   root, as opposed to retrieving it from the pre-image of the block in `latest_block`, reducing the merkle proof depth
+   and thus the cost of usage.
 
-[ExecutionPayload]: https://github.com/ethereum/consensus-specs/blob/dev/specs/bellatrix/beacon-chain.md#executionpayload
+1. The `withdrawal_storage_root` (`bytes32`) elevates the Merkle-Patricia-Trie ([MPT][g-mpt]) root of L2 Withdrawal
+   contract storage. Instead of a MPT proof to the Withdrawal contract account in the account trie, one can directly
+access the MPT storage trie root, thus reducing the verification cost of withdrawals on L1.
```

Comment:
Link to withdrawal contract spec.

Reply:
suggestion

> Instead of making an MPT proof for a withdrawal against the state root (proving first the storage root of the L2 withdrawal contract against the state root, then the withdrawal against that storage root), we can prove against the L2 withdrawal contract's storage root directly, thus reducing the verification cost of withdrawals on L1.